corelib/src/libs/SireMove/openmmfrenergyst.cpp

```diff
@@ -2834,47 +2834,52 @@ void OpenMMFrEnergyST::initialise()
 
     if (UseLink_flag == true)
     {
-        Molecule molecule = moleculegroup.moleculeAt(0).molecule();
 
-        bool haslinkinfo = molecule.hasProperty("linkbonds");
-
-        if (haslinkinfo)
+        for (int i = 0; i < nmols; i++)
```

Comment:
Do we really need to loop over all molecules to find the one with a `linksbonds` property? We know that this is the perturbable molecule, i.e. the one found in `OpenMMD.py` matching the `perturbed_residue_number`. We could simply pass through the molecule number to `openmmfrenergyst.cpp`, via the constructor or otherwise. That would involve unnecessary looping.

I would suggest adding an optional argument to the constructor that gives the number of the molecule, or perturbed residue, which could be something like -1 as default. If it's -1, then we can loop as you have, but if the value is set then we can just extract the molecule directly. This would involve rebuilding the Sire.Move wrappers since the API would have changed. (Although since the argument is optional, existing code won't break.)